### PR TITLE
feat(vertex): Add auth support in the vertexai

### PR DESCRIPTION
**Pull request by cynthiajoan**
## Description

Add the support for auth token in vertex request

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by Lyokone — APPROVED** [review]

File: packages/firebase_vertexai/firebase_vertexai/example/lib/main.dart [hunk 1]
```diff
@@ -526,7 +526,7 @@ class MessageWidget extends StatelessWidget {
             decoration: BoxDecoration(
               color: isFromUser
                   ? Theme.of(context).colorScheme.primaryContainer
-                  : Theme.of(context).colorScheme.surfaceContainerHighest,
+                  : Theme.of(context).colorScheme.surfaceVariant,
```
Comment (russellwheatley) [inline packages/firebase_vertexai/firebase_vertexai/example/lib/main.dart:529]: If you revert this, the analyse CI should successfully run again

**Review by russellwheatley — CHANGES_REQUESTED** [review]
analyse CI needs fixing: https://github.com/firebase/flutterfire/actions/runs/9132787893/job/25114851878?pr=12797#step:5:338

Other than that, LGTM.

File: packages/firebase_vertexai/firebase_vertexai/example/lib/main.dart (hunk 1 above)
Comment (cynthiajoan) [reply packages/firebase_vertexai/firebase_vertexai/example/lib/main.dart]: done

**Review by russellwheatley — APPROVED** [review]

**Comment by russellwheatley** [comment]
@cynthiajoan - is this going in after this PR? https://github.com/firebase/flutterfire/pull/12794

**Comment by natebosch** [comment]
> I'm thinking to have this in before that PR

Sounds good - I'll watch for this one to land and then resolve any merge conflicts.

**Pull request merged**